include/h2o.h

```diff
@@ -1127,6 +1127,10 @@ typedef void (*h2o_forward_datagram_cb)(h2o_req_t *req, h2o_iovec_t *datagrams,
 #define H2O_SEND_SERVER_TIMING_BASIC 1
 #define H2O_SEND_SERVER_TIMING_PROXY 2
 
+#define H2O_DATAGRAM_FORMAT_NONE     0
+#define H2O_DATAGRAM_FORMAT_DRAFT03  1
+#define H2O_DATAGRAM_FORMAT_RFC      2
```

Comment:
Actually I might have a different idea.

For the draft version, how about storing some non-standard string in `h2o_req_t::upgrade`. To give an example, it can be "connect-udp-draft04."

The reason IETF changed the way request is sent is because the draft version stuck out.

HTTP defines two ways of turning a HTTP channel into a bidirectional byte stream; one is the use of CONNECT method and the other is the use of the upgrade token (which is sent using the upgrade header or the :protocol pseudo header).

However, draft version of the masque protocol defined another way of turning the channel to a bidirectional bytestream, which was the CONNECT-UDP method.

That became a headache, because everybody had to adjust their HTTP parsing logic. Hence we decided to switch to the extension mechanism that we already have (i.e., use the upgrade token).

Maybe it would be a good idea to follow the decision ourselves.

Inside H2O, we would also rely on the value of `h2o_req_t::upgrade`. But for the draft version we have to retain code in each HTTP version. That can be kept as a logic that sets a special value into `h2o_req_t::upgrade`.

WDYT?

Reply:
That makes sense, thank you!